src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3, 1);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3 + 90.0f, 0);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3 + 90.0f, 1);
+            break;
+        }
+    }
+}
+
+void func_i2_80187FF8(Effect* effect, f32 x, f32 y, f32 z) {
+    Effect_Initialize(effect);
+    effect->obj.status = 1;
+    effect->obj.id = OBJ_EFFECT_371;
+
+    effect->obj.pos.x = x;
+    effect->obj.pos.y = y;
+    effect->obj.pos.z = z;
+
+    effect->unk_44 = 0x80;
+    effect->vel.z = 50.0f;
+    effect->scale2 = 1.0f;
+    Object_SetInfo(&effect->info, effect->obj.id);
+}
+
+void func_i2_80188088(Boss* boss) {
+    s32 i;
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187FF8(&gEffects[i], boss->obj.pos.x + 700.0f, boss->obj.pos.y, boss->obj.pos.z + 1235.0f);
+            Audio_PlaySfx(0x19006035U, &gEffects[i].sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187FF8(&gEffects[i], boss->obj.pos.x - 700.0f, boss->obj.pos.y, boss->obj.pos.z + 1235.0f);
+            break;
+        }
+    }
+}
+
+void func_i2_801881A8(Effect* effect, f32 x, f32 y, f32 z, f32 zRot, s32 arg5) {
+    Effect_Initialize(effect);
+    effect->obj.status = 1;
+    effect->obj.id = OBJ_EFFECT_370;
+
+    effect->obj.pos.x = x;
+    effect->obj.pos.y = y;
+    effect->obj.pos.z = z;
+
+    effect->obj.rot.z = zRot;
+
+    effect->unk_4E = arg5;
+    Object_SetInfo(&effect->info, effect->obj.id);
+}
+
+void func_i2_80188228(f32 x, f32 y, f32 z, f32 zRot) {
+    s32 i;
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_801881A8(&gEffects[i], x, y, z, zRot, 0);
+            break;
+        }
+    }
+}
+
+void func_i2_80188298(f32 x, f32 y, f32 z, f32 zRot) {
+    s32 i;
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_801881A8(&gEffects[i], x, y, z, zRot, -1);
+            Audio_PlaySfx(0x3102705CU, &gEffects[i].sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            break;
+        }
+    }
+}
+
+void func_i2_80188344(Boss* boss) {
+    Vec3f src;
+    Vec3f dest;
+    f32 temp;
+
+    if (boss->actionState < 5) {
+        boss->obj.pos.x = gBosses->obj.pos.x;
```

Comment:
```suggestion
        boss->obj.pos.x = gBosses[0].obj.pos.x;
```
Unless that doesn't match